astroid/rebuilder.py

```diff
@@ -1134,18 +1257,37 @@ def visit_decorators(
             return None
         # /!\ node is actually an _ast.FunctionDef node while
         # parent is an astroid.nodes.FunctionDef node
-        if PY38_PLUS:
+        if sys.version_info >= (3, 8):
```

Comment:
Huhu. I'm going to stop suggesting for every test then. But why though ? I'd say it's a good thing to use PY38+ if  only for the performance increase of not having to recalculate the bool everytime ?

Reply:
True, I don't really like the performance impact either. On the positive side `sys.version_info` is special cased by type checkers so they know how to infer it. That doesn't work with `PY38_PLUS`.

If performance is a real concern, the only other option, I think, would be to conditional define the `visit_xxx` methods itself. That would mean a lot of code duplications though. Not sure we could maintain that, at least not easily.

In the end, it's only the node tree. So the performance impact might be acceptable.

--
```py
    if sys.version_info >= (3, 8):
        def visit_constant(self, node: "ast.Constant", parent: NodeNG) -> nodes.Const:
            """visit a Constant node by returning a fresh instance of Const"""
            return nodes.Const(
                value=node.value,
                kind=node.kind,
                lineno=node.lineno,
                col_offset=node.col_offset,
                end_lineno=node.end_lineno,
                end_col_offset=node.end_col_offset,
                parent=parent,
            )
    else:
        def visit_constant(self, node: "ast.Constant", parent: NodeNG) -> nodes.Const:
            """visit a Constant node by returning a fresh instance of Const"""
            return nodes.Const(
                node.value,
                node.lineno,
                node.col_offset,
                parent,
                node.kind,
            )
```